### ISSUE TITLE
vmagent does not filter part of the metrics in metric_relabel_configs

### ISSUE THREAD

**Issue by Sinketsu** [VictoriaMetrics/VictoriaMetrics#4806]
### Describe the bug

We try filter our metrics from some endpoints with `if` expression (and action `drop`). Expected that all metrics that do match the regular expression will be deleted. However, some of the metrics still remain.
Perhaps it depends on the order of the timeseries, because in my example it affects.

### To Reproduce

Simple minimum example to reproduce:
1. Create some endpoint, which serves metrics in Prometheus format:
   ```
   # TYPE envoy_cluster_http2_header_overflow counter
   envoy_cluster_http2_header_overflow{} 1
   # TYPE envoy_http2_dropped_headers_with_underscores counter
   envoy_http2_dropped_headers_with_underscores{} 0
   ```
   I used text file with `python -m  http.server`
2. Start vmagent with scrape config:
   ```
   scrape_configs:
     - job_name: test
       scrape_interval: 30s
       scrape_timeout: 20s
       metrics_path: "/metrics"
       static_configs:
         - targets:
             - 'localhost:1234' # endpoint from 1 step
       metric_relabel_configs:
         - if: '{__name__!~"envoy_cluster_http2_header_overflow"}'
           action: drop
   ```
3. Start some receiver for remoteWrite requests for showing which data is sent to the storage. I used simple Go application, which prints timeseries in request (published to https://pastebin.com/1LtDzbrd)
4. Let's see which data is sent - both timeseries are present, although only one was expected
   In my case it prints:
   ```
   envoy_cluster_http2_header_overflow{instance=localhost:1234, job=test} 1.000000
   envoy_http2_dropped_headers_with_underscores{instance=localhost:1234, job=test} 0.000000
   up{instance=localhost:1234, job=test} 1.000000
   scrape_duration_seconds{instance=localhost:1234, job=test} 0.009000
   scrape_samples_scraped{instance=localhost:1234, job=test} 2.000000
   scrape_samples_post_metric_relabeling{instance=localhost:1234, job=test} 2.000000
   scrape_series_added{instance=localhost:1234, job=test} 2.000000
   scrape_timeout_seconds{instance=localhost:1234, job=test} 20.000000
   ```
5. Try change metrics set from 1 step to (reorder two series):
   ```
   # TYPE envoy_http2_dropped_headers_with_underscores counter
   envoy_http2_dropped_headers_with_underscores{} 0
   # TYPE envoy_cluster_http2_header_overflow counter
   envoy_cluster_http2_header_overflow{} 1
   ```
6. Let's see which data is sent - all works as expected
   In my case it prints:
   ```
   envoy_cluster_http2_header_overflow{instance=localhost:1234, job=test} 1.000000
   up{instance=localhost:1234, job=test} 1.000000
   scrape_duration_seconds{instance=localhost:1234, job=test} 0.020000
   scrape_samples_scraped{instance=localhost:1234, job=test} 2.000000
   scrape_samples_post_metric_relabeling{instance=localhost:1234, job=test} 1.000000
   scrape_series_added{instance=localhost:1234, job=test} 2.000000
   scrape_timeout_seconds{instance=localhost:1234, job=test} 20.000000
   ```

### Version

```
$ ./vmagent-prod --version
vmagent-20230728-101825-tags-v1.92.1-0-g8f257889c
```

### Logs

The logs that are related to this look ok:
```
2023-08-10T08:27:53.463Z	info	VictoriaMetrics/lib/promrelabel/if_expression.go:83	DEBUG: unmarshaling ifExpr from "{__name__!~\"envoy_cluster_http2_header_overflow\"}"
2023-08-10T08:27:53.464Z	info	VictoriaMetrics/lib/promrelabel/if_expression.go:92	DEBUG: unmarshaled ifExpr from "{__name__!~\"envoy_cluster_http2_header_overflow\"}" to {__name__!~"envoy_cluster_http2_header_overflow"}
2023-08-10T08:27:53.465Z	info	VictoriaMetrics/lib/promrelabel/if_expression.go:83	DEBUG: unmarshaling ifExpr from "{__name__!~\"envoy_cluster_http2_header_overflow\"}"
2023-08-10T08:27:53.465Z	info	VictoriaMetrics/lib/promrelabel/if_expression.go:92	DEBUG: unmarshaled ifExpr from "{__name__!~\"envoy_cluster_http2_header_overflow\"}" to {__name__!~"envoy_cluster_http2_header_overflow"}
```

### Used command-line flags

```
./vmagent-prod -promscrape.config=scrape-local.yml -remoteWrite.url=http://localhost:8080 -remoteWrite.forcePromProto=true -enableTCP6
```

### Additional information

Same behavior in 1.91.3

**Comment by valyala** [comment]
@Sinketsu , thanks for filing the bugreport! The issue should be fixed in the commit 1bd7637fe1ad361c6f63f711e08b081d979cb13d , which will be included in the next release. In the mean time you can build `vmagent` from this commit according to [these docs](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and verify whether `if` filter works properly now.

**Comment by valyala** [comment]
The fix for `if` filter in [relabeling rules](https://docs.victoriametrics.com/vmagent.html#relabeling-enhancements) should work properly starting from [v1.93.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as fixed.

**Comment by valyala** [comment]
FYI, this issue has been fixed also in [long-time support release v1.79.14](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.14).

**Comment by valyala** [comment]
FYI, the bugfix has been back-ported to the [long-term support release v1.87.7](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.7).
